tests/observatory/dags/telescopes/test_open_citations.py

```diff
@@ -46,6 +47,7 @@ def test_open_citations_release(self, mock_variable_get):
         """
 
         # Mock data variable
+        reset_variables()
```

Comment:
When do you need to use the 'reset_variables()' function?

Reply:
We won't need to use it once all of the telescopes are using the telescope template and the observatory test environment.

It is being used because we use global variables in [template_utils.py](https://github.com/The-Academic-Observatory/observatory-platform/blob/feature-geonames-telescope-template/observatory-platform/observatory/platform/utils/template_utils.py#L49-L54) so that the Google Cloud Secrets backend isn't called too much when requesting certain Airflow Variables. For these global variables, once the Airflow Variable has been fetched once, then it won't be fetched again, see [here](https://github.com/The-Academic-Observatory/observatory-platform/blob/feature-geonames-telescope-template/observatory-platform/observatory/platform/utils/template_utils.py#L83-L86) for an example. The problem when running unit tests, is that the first test to run which sets one of these global variables, will end up setting that value for all of the unit tests (as they won't update after that). When the tests run, the functions in template_utils.py need to be able to get the latest values for the Airflow Variables, which is why reset_variables() needs to be called before these tests are run.

These global variables are automatically reset in the observatory test environment, so it won't be a problem once we use that.

Reply:
Ah okay, thanks for the explanation!